applications/plugins/Communication/components/serverCommunicationVRPN.inl

```diff
@@ -0,0 +1,399 @@
+    if (subscribersMap.size() == 0)
+    {
+        if (isVerbose())
+            msg_info(this) << "Server Communication VRPN does not have Subscriber";
+        return;
+    }
+    for (std::map<std::string, CommunicationSubscriber*>::iterator it = subscribersMap.begin(); it != subscribersMap.end(); it++)
+    {
+        CommunicationSubscriber* subscriber = it->second;
+        std::string strTestText = subscriber->getSubject()+"@"+address;
+        const char *device = strTestText.c_str();
+        vrpn_text_sender = new vrpn_Text_Sender(device, m_connection);
+        vrpn_analog_server = new vrpn_Analog_Server(device, m_connection);
+        vrpn_tracker_server = new vrpn_Tracker_Server(device, m_connection);
+    }
+
+    while (!m_connection->connected() && this->m_running)
+        m_connection->mainloop();
+
+    while (m_connection->connected() && this->m_running)
+    {
+
+        for (std::map<std::string, CommunicationSubscriber*>::iterator it = subscribersMap.begin(); it != subscribersMap.end(); it++)
+        {
+            CommunicationSubscriber* subscriber = it->second;
+            ArgumentList argumentList = subscriber->getArgumentList();
+
+            try
+            {
+                for (ArgumentList::iterator itArgument = argumentList.begin(); itArgument != argumentList.end(); itArgument++ )
+                    createVRPNMessage(subscriber, *itArgument);
+            } catch(const std::exception& e) {
+                if (isVerbose())
+                    msg_info("ServerCommunicationVRPN") << e.what();
+            }
+            m_connection->mainloop(&delay);
+        }
+    }
+}
+
+void ServerCommunicationVRPN::createVRPNMessage(CommunicationSubscriber* subscriber, std::string argument)
+{
+    std::stringstream messageStr;
+    BaseData* data = fetchDataFromSenderBuffer(subscriber, argument);
+    if (!data)
+        throw std::invalid_argument("data is null");
+    const AbstractTypeInfo *typeinfo = data->getValueTypeInfo();
+    const void* valueVoidPtr = data->getValueVoidPtr();
+
+    delay.tv_sec = 0L;
+    delay.tv_usec = 0L;
+    angle += 0.001f;
+
+    if (!typeinfo->Container())
+    {
+        if (vrpn_text_sender)
+            vrpn_text_sender->send_message(data->getValueString().c_str(), vrpn_TEXT_NORMAL);
+
+        if (vrpn_analog_server)
+        {
+            vrpn_analog_server->setNumChannels(1);
+            double *channels = vrpn_analog_server->channels();
+            static int done = 0;
+            if (!done)
+            {
+                //converting string to double*
+                channels[0] = stod (data->getValueString());
+                done = 1;
+            }
+            else
+                channels[0] += stod (data->getValueString());
+
+            vrpn_analog_server->report_changes();
+        }
+
+        if(vrpn_tracker_server)
+        {   //Position of Tracker
+            pos[0] = sin ( angle );
+            pos[1] = stof (data->getValueString());
```

Comment:
same value for pos[1/2], does it works ?

Reply:
Yes, they work perfectly. No issues with the pos. I think the pos[0] can be allocated the same value.